### ISSUE TITLE
Error in postsolve

### ISSUE THREAD

**Issue by jajhall** [ERGO-Code/HiGHS#792]
A feasibility problem 
[phase1.mps.txt](https://github.com/ERGO-Code/HiGHS/files/8422359/phase1.mps.txt)
generated by the QP solver when solving `LASER` from Maros-Meszaros. 

The problem is reduced to empty, but then gets infinite values in the primal solution in postsolve

The first 1000 variables are free

**Comment by lgottwald** [comment]
This is an interesting case, I don't know yet how to fix. The infinite values do not come from a logical error, but from numerical blowup until overflow it seems.

The problem has weird properties. It contains 1000 equations which can be "factored" by presolve solely by substiution of free singleton columns creating a chain of dependencies of values. Eventually some free columns become empty columns and are hence set to value 0. This apparently leads to the assignment of values that are larger and larger as the values during postsolve blow up when undoing the free singleton substitutions.

The substitutions and fixings themselves all seem trivial and unproblematic by themselves and all values involved are numerically okay. All this happens in the first trivial presolve scan of rows and columns and no advanced presolve rule is used.

It seems to me as if there are restriction in the problem that are missing in the formulation. Clearly values blowing up way above 1e100 is not a feasible/desired solution to the problem. If there where some nonzero objective coefficients then maybe presolve would fix some of the empty columns not to zero but to 1 instead and arrive at a primal point with more reasonable values. If there where more primal column bounds then presolve would not be able to substitute all the columns as implied free columns unless the rows and bounds of other columns actually do imply the values. For the problem as it is given presolve could easily obtain a correct solution in the postsolve step when using arbitrary precision as the steps themselves involve numerically normal values and the solution values are what causes the blow up.

I wonder if the dropped values in the hessian play a role:
```
WARNING: Hessian matrix packed vector contains 229 |values| in [4.44089e-16, 1.13687e-13]
```

Maybe those would yield small nonzero cost coefficients to guide presolve in fixing empty columns to values that do not cause blow up.

**Comment by jajhall** [comment]
I guess you've looked at the behaviour of HiGHS on the original QP to have seen `WARNING: Hessian matrix packed vector contains 229 |values| in [4.44089e-16, 1.13687e-13]`. 

Michael just gives the LP without an objective to HiGHS and uses it to get a feasible point. I expect that there is a nontrivial Hessian that prevents the optimal solution from diverging too far. However, I can't see how, systematically, including an objective term will help. Indeed, presolve might well identify the LP as unbounded. 

Ultimately, QP presolve techniques may be needed.

**Comment by lgottwald** [comment]
Yes, for the free columns the objective can be problematic. There are however, several boxed columns and they often end up empty and are fixed to their lower bound, but maybe some should be pushed to their upper bound. The all zero solution for the free columns leads to blowup in the subsituted ones in this LP.

I guess the blowup could be prevented by giving the LP solver some more column bounds even if they are artificial. Bounding the columns by 1e4-1e6 should be fine in most cases, and only whenever that makes the QP infeasible the LP solver will provide the dual ray. Using the dual ray information any bounds that make the problem infeasible can simply be relaxed to make it feasible in one more pass.

**Comment by lgottwald** [comment]
Solving without presolve also works which I think is due to the numerical safe guards in the factorization. Probably HiGHS will choose different pivots due to their size or something like that. Because what presolve does is essentially just a factorization since most columns are free and there is always another free singleton column after removing the previous one with it's row.

**Comment by feldmeier** [comment]
Thanks a lot for the in-depth analysis! I'll try the trick on the whole testset. HiGHS isn't the only solver struggling with this instance, so it's quite a though one indeed
